lib/templates/src/main.js

```diff
@@ -1,10 +1,12 @@
 import Vue from 'vue';
 import VueMagpie from 'magpie-base';
+import store from './store/index';
 import App from './App.vue';
 
 Vue.config.productionTip = false;
 Vue.use(VueMagpie);
 
 new Vue({
+  store,
```

Comment:
@marcelklehr  Because of the requirement that whenever we want to use a `store`, we would need to put it inside of `new Vue({`, Therefore, I'm not sure if we can make the store part of the `magpie-base` package instead of including the store in the template. Maybe there's a way to do it via the plugin mechanism?

Reply:
In addition to that I realized after our session last week, that we block implementers from using vuex themselves, because magpie is already using it.

Perhaps we can make the vuex part a responsibility of the implementer and only expose hooks to the user. This would be similar to how `vue-socket.io` runs things (see here for a quick example code: https://www.digitalocean.com/community/tutorials/vuejs-vue-socketio#usage).

What do you think? I'm happy to help rewrite the code to make these changes, btw :)

Reply:
That's a nice example :) Yeah I'm not sure if most users will want to implement Vuex functionalities themselves, but it seems that we can have only one Vuex instance in the generated experiment app, so providing it as part of the package would not be practical. This seems like a reasonable workaround if it can be implemented easily.